sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -243,6 +243,18 @@ public static OffsetDateTime parseDate(String dateString) {
                     throw LOGGER.logExceptionAsError(new RuntimeException("I/O errors occurs. Error details: "
                         + e.getMessage()));
                 }
+            })
```

Comment:
Alternatively we can wrap the original `Flux` inside a `Flux.defer` and have 

```java
final long[] currentTotalLength = new long[1];
```

inside the defer which is some what more idiomatic I think.

Reply:
It will also match what we do in `RestProxy`. https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java#L146

Reply:
Good call. I was just working with what was already there, but I can clean that up.